3rdparty/mshadow/mshadow/tensor.h

```diff
@@ -69,15 +69,15 @@ struct Shape {
    * \param idx dimension index
    * \return the corresponding dimension size
    */
-  MSHADOW_XINLINE index_t &operator[](index_t idx) {
+  MSHADOW_XINLINE index_t &operator[](int idx) {
```

Comment:
Can you describe this change a bit more in detail ? Why is this required ?

Reply:
This is not directly fixing this bug. However, while checking the tensor struct, I noticed this data type should be int because it is indexing the dimension (which is set to int)